packages/axelar-wasm-std/src/msg_id/base_58_solana_event_index.rs

```diff
@@ -0,0 +1,362 @@
+use core::fmt;
+use std::{array::TryFromSliceError, fmt::Display, str::FromStr};
+
+use error_stack::{Report, ResultExt};
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use super::Error;
+use crate::nonempty;
+
+type RawSignature = [u8; 64];
+
+pub struct Base58SolanaTxDigestAndEventIndex {
+    // Base64 decoded bytes of the Solana signature.
+    pub signature: RawSignature,
+    pub event_index: u32,
+}
+
+impl Base58SolanaTxDigestAndEventIndex {
+    pub fn tx_digest_as_base58(&self) -> nonempty::String {
+        bs58::encode(self.signature)
+            .into_string()
+            .try_into()
+            .expect("failed to convert tx hash to non-empty string")
+    }
+
+    pub fn new(tx_id: impl Into<RawSignature>, event_index: impl Into<u32>) -> Self {
+        Self {
+            signature: tx_id.into(),
+            event_index: event_index.into(),
+        }
+    }
+
+    pub fn new_from_b58_encoded_signature_and_index(
```

Comment:
Added this extra constructor for convenience, and for not spreading the decoding code everywhre.